spikewrap/data_classes/base.py

```diff
@@ -0,0 +1,138 @@
+import fnmatch
+from collections import UserDict
+from collections.abc import ItemsView, KeysView, ValuesView
+from dataclasses import dataclass
+from itertools import chain
+from pathlib import Path
+from typing import Callable, Dict, List, Literal
+
+
+@dataclass
+class BaseUserDict(UserDict):
```

Comment:
Will `BaseUserDict` be inherited by another class different than `PreprocessingData`? I am not sure on why you need two classes. 
Then also `load_data()` adds another layer of abstraction, actually finalising the datastructure by filling in the `.data` attribute. Maybe this could be simplified.

Reply:
Later on there are `SortingData` class and it's subclasses that inhereit from this. However, we discussed the factory -method approach for `load_data` also. I will make a issue on this as OpenEphys support needs to be added before the first version release and there is somone interested now with test data.